yellowbrick/text/umap_vis.py

```diff
@@ -0,0 +1,351 @@
+    traits. For example, by passing ``cluster.labels_`` as ``y`` in ``fit()``, all
+    points in the same cluster will be grouped together. This extends the
+    neighbor embedding with more information about similarity, and can allow
+    better interpretation of both clusters and classes.
+
+    The current default for UMAP is Euclidean distance.  Hellinger distance would be a more appropriate distance
+    function to use with CountVectorize data.  That will be released in the next version of UMAP.
+    In the meantime cosine distance is likely a better text default that Euclidean and can be set using
+    the keyword argument metric='cosine'.
+
+    For more, see https://github.com/lmcinnes/umap
+
+    Parameters
+    ----------
+
+    ax : matplotlib axes
+        The axes to plot the figure on.
+
+    labels : list of strings
+        The names of the classes in the target, used to create a legend.
+        Labels must match names of classes in sorted order.
+
+    colors : list or tuple of colors
+        Specify the colors for each individual class
+
+    colormap : string or matplotlib cmap
+        Sequential colormap for continuous target
+
+    random_state : int, RandomState instance or None, optional, default: None
+        If int, random_state is the seed used by the random number generator;
+        If RandomState instance, random_state is the random number generator;
+        If None, the random number generator is the RandomState instance used
+        by np.random. The random state is applied to the preliminary
+        decomposition as well as UMAP.
+
+    alpha : float, default: 0.7
+        Specify a transparency where 1 is completely opaque and 0 is completely
+        transparent. This property makes densely clustered points more visible.
+
+    kwargs : dict
+        Pass any additional keyword arguments to the UMAP transformer.
+
+
+    Examples
+    --------
+
+    >>> model = MyVisualizer(metric='cosine')
+    >>> model.fit(X)
+    >>> model.poof()
+
+    """
+
+    # NOTE: cannot be np.nan
+    NULL_CLASS = None
+
+    def __init__(self, ax=None,
+                 labels=None, classes=None, colors=None, colormap=None,
+                 random_state=None, alpha=0.7, **kwargs):
+
+        # Visual Parameters
+        self.alpha = alpha
+        self.labels = labels
+        self.colors = colors
+        self.colormap = colormap
+        self.random_state = random_state
+
+
+        # Fetch UMAP kwargs from kwargs by popping only keys belonging to UMAP params
+        umap_kwargs = {
+            key: kwargs.pop(key)
+            for key in UMAP().get_params()
+            if key in kwargs
+        }
+
+        self.transformer_ = self.make_transformer(umap_kwargs)
+
+        # Call super at the end so that size and title are set correctly
+        super(UMAPVisualizer, self).__init__(ax=ax, **kwargs)
+
+    def make_transformer(self, umap_kwargs={}):
```

Comment:
I have a preference for leaving it in if you are OK with that.  It provides some nice consistency between the two classes and makes the code work as a nice template in case one wants to extend or modify the classes with any extra transformations at a later date.

Reply:
That's good reasoning and I'm willing to go with it - we'll keep it in!